### DISCUSSION TITLE
[Blog] Utility-Friendly Transitions with @tailwindui/react

### DISCUSSION THREAD

**Discussion by adamwathan** [tailwindlabs/tailwindcss#2262] [Show and tell]
Earlier this week, we released the very first version of [@tailwindui/react](https://github.com/tailwindlabs/tailwindui-react), a library that provides low-level primitives for turning utility-first HTML into fully interactive UIs.

We'll be adding many more components in the coming months _(like dropdowns, toggles, modals, and more, and for Vue too!)_ but thought we'd start with a `<Transition>` component to at least get the current Tailwind UI experience for React users up to par with what's possible in Vue and Alpine.js.

Here's what it looks like to use:

```jsx
import { Transition } from '@tailwindui/react'
import { useState } from 'react'

function MyComponent() {
  const [isOpen, setIsOpen] = useState(false)

  return (
    <div>
      <button onClick={() => setIsOpen(!isOpen)}>Toggle</button>
      <Transition
        show={isOpen}
        enter="transition-opacity duration-75"
        enterFrom="opacity-0"
        enterTo="opacity-100"
        leave="transition-opacity duration-150"
        leaveFrom="opacity-100"
        leaveTo="opacity-0"
      >
        {/* Will fade in and out */}
      </Transition>
    </div>
  )
}
```

[Check out the blog post](https://blog.tailwindcss.com/utility-friendly-transitions-with-tailwindui-react) for more and let us know what you think! 🙌

**Comment by runoncedev** [comment]
Maybe the post should be updated since `@tailwindui/react` has been deprecated in favor of `@headlessui/react`.
